### ISSUE TITLE
TemplateRenderer's Event Handler not working inside secondary Tabs

### ISSUE THREAD

**Issue by tarekoraby** [vaadin/flow#9754]
### Description of the bug / feature
I have a view with two Tab components, each is displaying an identical Grid with a column constructed using a TemplateRenderer. Adding an event handler to those TemplateRenderers only works for the Grid in the first Tab (i.e. the Tab that was initially rendered). It doesn't work for the second one. 

This issue only occurs in version 14.4.5 (the event-handling works fine in 14.4.4).

### Minimal reproducible example
```
@Route("HelloWorldView")
public class HelloWorldView extends VerticalLayout {

    Grid<String> grid1;
    Grid<String> grid2;

    public HelloWorldView() {
        grid1 = addGrid();
        grid2 = addGrid();

        Tab tab1 = new Tab("1");
        Tab tab2 = new Tab("2");
        Tabs tabs = new Tabs(tab1, tab2);

        tabs.addSelectedChangeListener(this::doChange);
        grid1.setVisible(true);
        grid2.setVisible(false);
        add(tabs, grid1, grid2);
    }

    private void doChange(Tabs.SelectedChangeEvent t) {
        if (!t.isFromClient()) {
            return;
        }

        grid1.setVisible(false);
        grid2.setVisible(false);

        if (t.getSelectedTab().getLabel().equalsIgnoreCase("1")) {
            grid1.setVisible(true);
        }
        if (t.getSelectedTab().getLabel().equalsIgnoreCase("2")) {
            grid2.setVisible(true);
        }
    }

    Grid<String> addGrid() {
        Grid<String> grid = new Grid<>();
        grid.setSelectionMode(Grid.SelectionMode.NONE);

        Icon newUserIcon = VaadinIcon.PLUS.create();
        newUserIcon.addClickListener(c -> Notification.show(c.toString()));

        grid.addColumn(TemplateRenderer.<String> of(
                "<div on-click='edit'>content</div>").withEventHandler("edit",
                        Notification::show)
        ).setHeader(newUserIcon).setTextAlign(ColumnTextAlign.START);

        grid.setItems(Arrays.asList("1", "2", " 3"));
        return grid;
    }

}
```

Running this view:

1.  open the first tab -> click on the rendered div elements
2.  open the second tab -> click on the rendered div elements

### Expected behavior
The click events on both grids are handled (with a Notification shown)

### Actual behavior
Only the Grid in the first Tab has its event handled ( not sure if they are event fired) 

### Versions:
    - Vaadin / Flow version: 14.4.5
    - Java version: 11
    - OS version: Windows
    - Browser version: Chrome & FF

**Comment by pleku** [comment]
Just for the record, I don't see anything obvious in https://github.com/vaadin/platform/releases/14.4.5 that could cause a regression for this. But needs to be investigated.

**Comment by tanbt** [comment]
This is fixed in `14.4.6`, probably related to version conflict between frontend components.
@tarekoraby Please double check if it works for you.

**Comment by tarekoraby** [comment]
@tanbt confirmed! Seems to be working in 14.4.6. Thanks!
